ci/stage/elasticsearch_s3backup.Jenkinsfile

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env groovy
+/**
+  This is the Jenkinsfile for backing up elasticsearch using s3.
+  Elasticsearch S3 repository plugin is required.
+ */
+ansiColor('xterm') {
+  timestamps {
+        node('dhslave') {
+          configFileProvider(
+          [configFile( fileId: 'kubeconfig', variable: 'KUBECONFIG')]){
+	      stage('Pull Git Repositories') {
+                checkout_playbook_repo()
+              }
+              stage('Trigger Playbook') {
+                start_backup()
+              }
+          }
+        }
+  }
+}
+def checkout_playbook_repo() {
+    checkout poll: false, scm: [
+      $class: 'GitSCM',
+      branches: [[name: "*/master"]],
+      doGenerateSubmoduleConfigurations: false,
+      extensions: [
+        [$class: 'WipeWorkspace'],
+        [$class: 'RelativeTargetDirectory', relativeTargetDir: 'dh-ci-util']
+      ],
+      submoduleCfg: [],
+      userRemoteConfigs: [[url: 'https://gitlab.cee.redhat.com/data-hub/dh-ci-util.git']]
+    ]
+}
+def start_backup() {
+  try {
+    // Check if repository exists
+    checkRepositoryResponse = sh (
+      script: 'curl --insecure --cert $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.crt --key $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.key -i -k ${ELASTICSEARCH_ENDPOINT}/_snapshot/${BACKUP_NAME}',
+      returnStdout: true
+    ).trim()
+    
+    repoDoesNotExist = checkRepositoryResponse.toLowerCase().contains('404 not found')
+    
+    // Create the repository if one does not exist
+    if (repoDoesNotExist)
+    {
+        createRepositoryResponse = sh (
+            script: 'curl --insecure --cert $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.crt --key $WORKSPACE/dh-ci-util/certificates/stage-es-certs/system.admin.key -X PUT -H "Content-Type: application/json" -d ' + /'{"type": "s3","settings": {"bucket": "${S3_BUCKET}","base_path": "${S3_PATH}","endpoint": "${S3_ENDPOINT}","access_key":"${S3_ACCESS_KEY}","secret_key":"${S3_SECRET_KEY}"}}'/ + ' -i ${ELASTICSEARCH_ENDPOINT}/_snapshot/${BACKUP_NAME}',
```

Comment:
also where are all the env vars coming from?

Reply:
Which environment variables? The S3 ones are going to be passed in via Jenkins.  There will be default values for them.